xds/src/main/java/com/linecorp/armeria/xds/XdsConverterUtil.java

```diff
@@ -52,13 +78,21 @@ static List<Endpoint> convertEndpoints(ClusterLoadAssignment clusterLoadAssignme
                             } else {
                                 return Endpoint.of(socketAddress.getAddress(), socketAddress.getPortValue());
                             }
-                        })).collect(Collectors.toList());
+                        })).collect(toImmutableList());
     }
 
-    static List<Endpoint> convertEndpoints(Collection<ClusterLoadAssignment> clusterLoadAssignments) {
-        return clusterLoadAssignments.stream()
-                                     .flatMap(cla -> convertEndpoints(cla).stream())
-                                     .collect(Collectors.toList());
+    private static boolean containsExactlyInAnyOrder(Struct metadata1, Struct metadata2) {
+        if (metadata1.getFieldsCount() != metadata2.getFieldsCount()) {
```

Comment:
Question) Why is this condition necessary?

My understanding was that an endpoint is allowed to have more fields than the selector fields.

e.g.

- routeMatcher: `(key1: value1)`
- selector: `key1`
- endpoints
  - endpoint1: `(key1: value1), (key2:value2)`
  - endpoint2: `(key1: value1)`
-> in this case, I expected both `endpoint1` and `endpoint2` to match

Reply:
Oops, my bad. 😓